cmd/sqlflowserver/main_test.go

```diff
@@ -2,26 +2,95 @@ package main
 
 import (
 	"context"
+	"io"
 	"log"
+	"net"
 	"testing"
 	"time"
 
 	"google.golang.org/grpc"
 
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/golang/protobuf/ptypes/wrappers"
 	pb "github.com/sql-machine-learning/sqlflow/server/proto"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestRun(t *testing.T) {
-	a := assert.New(t)
-	tests := []string{
-		"show databases;",
-		"select * from iris.train limit 2;",
+func WaitPortReady(addr string, timeout time.Duration) {
+	// Set default timeout to
+	if timeout == 0 {
+		timeout = time.Duration(1) * time.Second
+	}
+	for {
+		conn, err := net.DialTimeout("tcp", addr, timeout)
+		if err != nil {
+			log.Printf("%s, try again", err.Error())
+		}
+		if conn != nil {
+			err = conn.Close()
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+}
+
+func AssertEqualAny(a *assert.Assertions, expected interface{}, actual *any.Any) {
```

Comment:
Thanks @tonyyang-svail for sample code here, it helped me alot: https://github.com/typhoonzero/sqlflow/pull/1/files